deepprofiler/learning/training.py

```diff
@@ -36,6 +49,14 @@ def learn_model(config, dset, epoch):
     configuration = tf.ConfigProto()
     configuration.gpu_options.visible_device_list = "0"
     main_session = tf.Session(config = configuration)
+    val_session = tf.Session(config = configuration)
+    keras.backend.set_session(val_session)
+    val_crop_generator.start(val_session)
+    x_validation, y_validation = deepprofiler.learning.validation.validate(
+            config,
```

Comment:
No need to pass the full `config` object to this function. Replace with a `top_k` parameter in the function.

Reply:
Same with the other calls in this file.